nebula_hw_interfaces/include/nebula_hw_interfaces/nebula_hw_interfaces_continental/continental_srr520_hw_interface.hpp

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NEBULA_CONTINENTAL_SRR520_HW_INTERFACE_H
+#define NEBULA_CONTINENTAL_SRR520_HW_INTERFACE_H
+
+#include <nebula_common/continental/continental_srr520.hpp>
+#include <nebula_hw_interfaces/nebula_hw_interfaces_common/nebula_hw_interface_base.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <ros2_socketcan/socket_can_receiver.hpp>
+#include <ros2_socketcan/socket_can_sender.hpp>
+
+#include <nebula_msgs/msg/nebula_packet.hpp>
+#include <nebula_msgs/msg/nebula_packets.hpp>
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+namespace nebula
+{
+namespace drivers
+{
+namespace continental_srr520
+{
+/// @brief Hardware interface of the Continental SRR520 radar
+class ContinentalSRR520HwInterface
+{
+public:
+  /// @brief Constructor
+  ContinentalSRR520HwInterface();
+
+  /// @brief Starting the interface that handles UDP streams
+  /// @return Resulting status
+  Status SensorInterfaceStart();
+
+  /// @brief Function for stopping the interface that handles UDP streams
+  /// @return Resulting status
+  Status SensorInterfaceStop();
+
+  /// @brief Setting sensor configuration
+  /// @param sensor_configuration SensorConfiguration for this interface
+  /// @return Resulting status
+  Status SetSensorConfiguration(
+    const std::shared_ptr<
+      const nebula::drivers::continental_srr520::ContinentalSRR520SensorConfiguration>
+      new_config_ptr);
+
+  /// @brief Registering callback for PandarScan
+  /// @param scan_callback Callback function
+  /// @return Resulting status
+  Status RegisterPacketCallback(
+    std::function<void(std::unique_ptr<nebula_msgs::msg::NebulaPacket>)> packet_callback);
+
+  /// @brief Sensor synchronization routine
+  void SensorSync();
+
+  /// @brief Process a new Sync Fup request
```

Comment:
Clearer doc comment
```suggestion
  /// @brief Process a new Sync Follow-up request
```

Reply:
Instead of this I changed fup mentions in the whole code just in case
[46dde25](https://github.com/tier4/nebula/pull/124/commits/46dde25f3a0e8fcf46c1995d3526283dcccb9178)